extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelProcessor.java

```diff
@@ -431,6 +432,37 @@ CamelComponentNameResolverBuildItem componentNameResolver(
         return new CamelComponentNameResolverBuildItem(recorder.createComponentNameResolver(componentNames));
     }
 
+    /**
+     * Discovers classes annotated with @DataTypeTransformer for package scanning and reflection.
+     * This enables transformer.scan() to work in native mode.
+     */
+    @BuildStep
+    void discoverDataTypeTransformers(
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<CamelPackageScanClassBuildItem> packageScanClass,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+
+        IndexView view = combinedIndex.getIndex();
+        DotName dataTypeTransformer = DotName.createSimple("org.apache.camel.spi.DataTypeTransformer");
+
+        Set<String> transformerClasses = view.getAnnotations(dataTypeTransformer)
+                .stream()
+                .filter(ai -> ai.target().kind() == AnnotationTarget.Kind.CLASS)
+                .map(ai -> ai.target().asClass().name().toString())
+                .collect(Collectors.toSet());
+
+        if (!transformerClasses.isEmpty()) {
+            LOGGER.debug("Found @DataTypeTransformer annotated classes: {}", transformerClasses);
+            packageScanClass.produce(new CamelPackageScanClassBuildItem(transformerClasses));
+
+            // Register transformer classes for reflection so they can be instantiated at runtime
+            transformerClasses.forEach(className -> reflectiveClass.produce(
+                    ReflectiveClassBuildItem.builder(className)
+                            .methods()
```

Comment:
Nitpick - if Camel does not need reflective access to any methods on the transformer classes, then you can remove this. It may be enough to only enable constructor reflection (enabled by default with `ReflectiveClassBuildItem`).